src/overlays/actors/ovl_Bg_Ikana_Mirror/z_bg_ikana_mirror.c

```diff
@@ -28,99 +30,365 @@ const ActorInit Bg_Ikana_Mirror_InitVars = {
+    }
+
+    for (i = 0; i < ARRAY_COUNT(this->colliderQuad); i++) {
+        Collider_InitQuad(play, &this->colliderQuad[i]);
+        Collider_SetQuad(play, &this->colliderQuad[i], &this->dyna.actor, &sQuadInit[i]);
+    }
+
+    BgIkanaMirror_SetQuadVertices(this);
+    this->lightAbsorptionAnimatedTexture = Lib_SegmentedToVirtual(&gStoneTowerTempleMirrorLightAbsorptionTexAnim);
+    this->lightEmissionAnimatedTexture = Lib_SegmentedToVirtual(&gStoneTowerTempleMirrorLightEmissionTexAnim);
+    BgIkanaMirror_SetupWait(this);
+}
+
+void BgIkanaMirror_Destroy(Actor* thisx, PlayState* play) {
+    BgIkanaMirror* this = THIS;
+    s32 i;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyTris(play, &this->colliderTris);
+
+    for (i = 0; i < ARRAY_COUNT(this->colliderQuad); i++) {
+        Collider_DestroyQuad(play, &this->colliderQuad[i]);
+    }
+}
+
+void BgIkanaMirror_SetupWait(BgIkanaMirror* this) {
+    this->isEmittingLight = 0;
+    this->actionFunc = BgIkanaMirror_Wait;
+}
+
+void BgIkanaMirror_Wait(BgIkanaMirror* this, PlayState* play) {
+    s8 isEmittingLight;
+    s32 startEmittingLight = false;
+
+    // The light emission texture should gradually disappear from sight.
+    if (this->lightEmissionAlpha >= 101) {
+        this->lightEmissionAlpha -= 100;
+    } else {
+        this->lightEmissionAlpha = 0;
+    }
+
+    // This checks if light is touching the mirror.
+    if (this->colliderTris.base.acFlags & AC_HIT) {
+        this->colliderTris.base.acFlags &= ~AC_HIT;
+        this->isEmittingLight = 0;
+        if (this->timer < 400) {
+            this->timer++;
+        }
+        if (this->lightAbsorptionAlpha < 195) {
+            this->lightAbsorptionAlpha += 60;
+        } else {
+            this->lightAbsorptionAlpha = 255;
+        }
+    } else {
+        isEmittingLight = this->isEmittingLight;
+        if (isEmittingLight > 0) {
+            startEmittingLight = true;
+        } else if (this->timer > 0) {
+            this->isEmittingLight = isEmittingLight + 1;
+        }
+    }
+
+    if (startEmittingLight) {
+        BgIkanaMirror_SetupEmitLight(this);
+    } else {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->colliderTris.base);
+    }
+}
+
+void BgIkanaMirror_SetupEmitLight(BgIkanaMirror* this) {
+    this->dyna.actor.flags |= ACTOR_FLAG_20;
+    this->actionFunc = BgIkanaMirror_EmitLight;
+}
+
+void BgIkanaMirror_EmitLight(BgIkanaMirror* this, PlayState* play) {
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(this->colliderQuad); i++) {
+        if ((this->colliderQuad[i].base.atFlags & AT_HIT)) {
+            this->colliderQuad[i].base.atFlags &= ~AT_HIT;
+        }
+    }
+
+    if (this->lightEmissionAlpha < 155) {
+        this->lightEmissionAlpha += 100;
+    } else {
+        this->lightEmissionAlpha = 255;
+    }
+
+    if (this->lightAbsorptionAlpha >= 61) {
```

Comment:
```suggestion
    if (this->lightAbsorptionAlpha > 60) {
```
makes more sense in the surrounding context.

Reply:
Fixed in the next commit to be pushed.